### DISCUSSION TITLE
TypeError: UniverPlugin is not a constructor

### DISCUSSION THREAD

**Discussion by aniruddha-k** [dream-num/univer#2526] [Q&A | 常见问题] [answered]
Hi Everyone,

I am encountering an issue while using Webpack with the UniverPlugin. When I attempt to utilize the UniverPlugin, I receive the following error:

TypeError: UniverPlugin is not a constructor

I am currently using the import statement to require the plugin, other plugins have 

Any assistance in resolving this issue would be greatly appreciated.

Webpack version: ^5.73.0
UniverPlugin version: ^0.4.3

Thank you!

**Comment by hexf00** [comment]
You can find the usage of `@univerjs/webpack-plugin` with Webpack 4 [here](https://univer.ai/playground?title=Webpack+4). The error you're encountering is likely due to TypeScript's type checking, as the plugin does not export its types. However, this error won't affect the actual functionality of the plugin, so you can safely ignore it.

As a temporary workaround, you can redefine the types in your project with:

```ts
declare module "@univerjs/webpack-plugin" {
  const UniverPlugin: any;
}
```

We will export the types in future versions.

**Reply by aniruddha-k** [reply]
Hi @hexf00, 

Thank you for replying.

I've added the above types in our project and also checked Webpack 4 plugin, now its not giving the above error.

But the code is getting stuck here (as shown in the image.)
![image](https://github.com/dream-num/univer/assets/16478831/e2210319-bf99-4086-9d72-7e68e4c922c7)

Do you have any other solution? I'm using Module Federation.

Thank you.

**Reply by hexf00** [reply]
Could you provide a demo that reproduces this issue, so that we can investigate it further?

**Comment by aniruddha-k** [comment, marked as answer]
Found the solution, the error was due to the style of importing the 'univer:locales' which was causing the error.

Referred to the documentation and used the below code instead of 'univer:locales'

```
import { LocaleType, Tools } from '@univerjs/core';
import DesignEnUS from '@univerjs/design/locale/en-US';
import UIEnUS from '@univerjs/ui/locale/en-US';
import DocsUIEnUS from '@univerjs/docs-ui/locale/en-US';
import SheetsEnUS from '@univerjs/sheets/locale/en-US';
import SheetsUIEnUS from '@univerjs/sheets-ui/locale/en-US';
 
const univer = new Univer({
  theme: defaultTheme,
  locale: LocaleType.EN_US,
  locales: {
    [LocaleType.EN_US]: Tools.deepMerge(
      SheetsEnUS,
      DocsUIEnUS,
      SheetsUIEnUS,
      UIEnUS,
      DesignEnUS,
    ),
  },
});
```
